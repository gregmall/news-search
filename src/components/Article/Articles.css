.Articles {
 
    list-style-type: none;
  
  }