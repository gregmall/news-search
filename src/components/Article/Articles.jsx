import React from 'react';
import styles from './Articles.css'
const Articles = ({articles}) =>{
  
  const articlesElements = articles.map(article => (

    <li key={article.name} className={styles.Articles}>
       <h2>{article.title}</h2>
       <h4>{article.author}</h4>
       <p>{article.description}</p>
       <span><a href={article.url} target="blank">{article.url}</a></span>
    </li>

  ));
  return <ul> {articlesElements} </ul>
  
}
export default Articles;





