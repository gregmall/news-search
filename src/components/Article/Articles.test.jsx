import React from 'react';
import { render, cleanup } from '@testing-library/react'
import Articles from './Articles';

describe('Articles component', () => {
    afterEach(() => cleanup());
    it('renders Article', () => {
        const { asFragment } = render(<Articles
          articles={[  
            {
              title: "Test Article",
              author: "Test Author",
              description: "Test Description",
              url: "Test Url"
            }
        ]}
      />);
        expect(asFragment()).toMatchSnapshot()
    });
});
