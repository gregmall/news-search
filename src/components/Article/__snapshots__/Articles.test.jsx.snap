// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component renders Article 1`] = `
<DocumentFragment>
  <ul>
     
    <li
      class="Articles"
    >
      <h2>
        Test Article
      </h2>
      <h4>
        Test Author
      </h4>
      <p>
        Test Description
      </p>
      <span>
        <a
          href="Test Url"
          target="blank"
        >
          Test Url
        </a>
      </span>
    </li>
     
  </ul>
</DocumentFragment>
`;
